src/conn.c

```diff
@@ -2139,9 +2139,14 @@ _evStopPolling(natsConnection *nc)
 static bool
 _processOpError(natsConnection *nc, natsStatus s, bool initialConnect)
 {
+    bool forceReconnect;
+
     natsConn_Lock(nc);
 
-    if (!initialConnect)
+    forceReconnect = initialConnect || nc->forceReconnect;
+    nc->forceReconnect = false;
+
+    if (!forceReconnect)
```

Comment:
Still, even if we override `allowReconnect`, I don't see why here we would change this line because the check of `_isConnecting()`, `natsConn_isClosed()`, etc.. is still valid. Meaning that we should not attempt to reconnect if any of these conditions are true.

Reply:
It's the next check, where we wouldn't attempt to reconnect unless allowReconnect is set. A false negative (vs. a false positive).

Reply:
But that's my point. If allowReconnect is false, then `natsConnection_Reconnect()` should check for that and return an error (and not close the socket obviously). But that's something that needs to be discussed/agreed with the rest of the team.

Reply:
One thing that the Go client is doing that we may want to do (and so, we may need some information that we are in the process of a "force reconnect") is that we could skip the wait that we have when attempting to reconnect. The "force reconnect" flag should then be reset only in the "doReconnect" function, when successfully reconnected, and likely checked in `natsConnection_Reconnect()` to be a no-op if that flag is currently set to true (which means that it would still be in progress).

Reply:
yes, done. I'll double-check this semantics with @piotrpio and @Jarema.